vortex-datafusion/src/persistent/format.rs

```diff
@@ -159,9 +162,16 @@ impl FileFormat for VortexFormat {
         }
     }
 
+    fn file_source(&self) -> Arc<dyn FileSource> {
+        Arc::new(VortexSource::new(
+            self.footer_cache.clone(),
+            VortexExecMetrics::default(),
```

Comment:
we are creating a new metrics container per call but as far as I see this is only used by datafusion for downcasting, so it should be fine